src/ZPublisher/BeforeTraverse.py

```diff
@@ -120,7 +124,11 @@ class NameCaller(object):
     >>> registerBeforeTraverse(folder, NameCaller('preop'), 'XApp')
     """
 
-    def __init__(self, name):
+    def __init__(self, name=None):
```

Comment:
Another case for an explicit string, rather than `None`:  maybe `'<undefined name>'`?